### PR TITLE
Problem: Relayer incentivization is not enabled in `chain-main`

### PR DESCRIPTION
**Pull request by devashishdxt**
Added relayer incentivization code in `app.go`. Related to #765.

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -357,14 +365,26 @@ func New(
 		),
 	)
 
+	// IBC Fee Module keeper
+	app.IBCFeeKeeper = ibcfeekeeper.NewKeeper(
+		appCodec, keys[ibcfeetypes.StoreKey], app.GetSubspace(ibcfeetypes.ModuleName),
+		app.IBCFeeKeeper, // may be replaced with IBC middleware
```
Comment (tomtau) [inline app/app.go:371]: so IBC middleware isn't there yet?

File: go.mod [hunk 2]
```diff
@@ -4,8 +4,8 @@ go 1.17
 
 require (
 	github.com/confluentinc/bincover v0.1.0
-	github.com/cosmos/cosmos-sdk v0.45.3
-	github.com/cosmos/ibc-go/v3 v3.0.0
+	github.com/cosmos/cosmos-sdk v0.45.4
+	github.com/cosmos/ibc-go/v3 v3.0.0-alpha1.0.20220510152759-dcd0681d8f07
```
Comment (tomtau) [inline go.mod:8]: I guess one can't put `ics29-beta2` as the version string?

File: x/icaauth/module_ibc.go [hunk 3]
```diff
@@ -38,13 +40,17 @@ func (am IBCModule) OnChanOpenInit(
 	chanCap *capabilitytypes.Capability,
 	counterparty channeltypes.Counterparty,
 	version string,
-) error {
+) (string, error) {
```
Comment (tomtau) [inline x/icaauth/module_ibc.go:43]: is icaauth ok with future channel upgrades?

**Review by tomtau — APPROVED** [review]
looks all right; one thing to consider is if we know how that channel upgrade functionality will work and whether there's anything to change in ICA auth to prepare it for it

File: app/app.go (hunk 1 above)
Comment (devashishdxt) [reply app/app.go:371]: We can add more middlewares in the future. I'll modify the comment.

File: go.mod (hunk 2 above)
Comment (devashishdxt) [reply go.mod:8]: Correct. It's just a version tag and not a release, so, `go mod tidy` automatically replaces it with this version.

File: x/icaauth/module_ibc.go (hunk 3 above)
Comment (devashishdxt) [reply x/icaauth/module_ibc.go:43]: I don't think there are any changes required in `chain-main`. But, I think, we'll need some changes in `ibc-go` ([here](https://github.com/cosmos/ibc-go/blob/9f70a070d773f8bfdb62d4205d8878f3149f351a/modules/apps/27-interchain-accounts/controller/keeper/account.go#L45)). I've asked Carlos about this.

**Pull request merged**